llvm/lib/CodeGen/PreISelIntrinsicLowering.cpp

```diff
@@ -233,6 +234,60 @@ static bool canEmitLibcall(const TargetMachine *TM, Function *F,
   return TLI->getLibcallName(LC) != nullptr;
 }
 
+// Return a value appropriate for use with the memset_pattern16 libcall, if
+// possible and if we know how. (Adapted from equivalent helper in
+// LoopIdiomRecognize).
+static Constant *getMemSetPattern16Value(MemSetPatternInst *Inst,
+                                         const TargetLibraryInfo &TLI) {
+  // FIXME: This could check for UndefValue because it can be merged into any
```

Comment:
Replace FIXME w/TODO.  Usually TODO means there's a bug, not a missed optimization.

Reply:
Thanks, this is another item inherited from the original version of the function in LoopIdiomRecognize.